charmcraft/helptexts.py

```diff
@@ -33,6 +33,24 @@
 # XXX Facundo 2020-09-10: we should add an extra (separated) line to the summary with:
 #   See <url> for additional documentation.
 
+# generic intro and outro texts
+HEADER = """
+Usage:
+    charmcraft [help] <command>
+"""
+
+USAGE = """\
+Usage: charmcraft [OPTIONS] COMMAND [ARGS]...
+Try '{fullcommand} -h' for help.
+
+Error: {error_message}
+"""
+
+
+def get_usage_message(fullcommand, error_message):
+    """Build a usage and error message."""
+    return USAGE.format(fullcommand=fullcommand, error_message=error_message)
+
```

Comment:
It seems a bit weird to me to have usage only be usable for errors. I would do something like:
```suggestion
USAGE = """\
￼Usage: charmcraft [OPTIONS] COMMAND [ARGS]...
￼Try '{fullcommand} -h' for help.
￼"""

ERROR_MSG = """\
{usage}

Error: {error_message}
"""
￼
def get_usage_message(fullcommand):
    """Build a usage message"""
    return USAGE.format(fullcommand=fullcommand)￼

￼def get_error_message(fullcommand, error_message):
￼    """Build an error message."""
￼    return ERROR_MSG.format(usage=get_usage_message(fullcommand=fullcommand), error_message=error_message)
```